app/views/posts/show.html.erb

```diff
@@ -2,15 +2,19 @@
 <div class="post">
   <div class="my-post">
     <div class='post-title'>
+    
       <h3><%=@posts.title%> | by <%= @posts.author.name%></h3>
       <p class="stats-comment"> Comments: <%= @posts.comments_counter or 0 %> | Likes: <%= @posts.likes_counter or 0 %> </p>
     </div>
       <p>Post : <%= @posts.text%> </p>
   </div>
   <div class="comment">
-    <% @posts.comments.each do |comment| %>
-      <% username = User.find(comment.author_id).name %>
+     <% @post.comments.includes([:author, :post]).each do |comment| %>
+  <h1><% username = User.find(comment.author_id).name %></h1>
       <p> <%= username%>: <%= comment.text%><p>
+      <% if can? :destroy, Comment %>
+       <%= button_to 'delete comment', user_post_comment_path(comment.author, comment.post,comment), method: :delete %>
+    <% end %>
```

Comment:
[OPTIONAL] I think it would be best to enhance the user experience by making the delete button only visible to users who are authorized to delete. 

- This can avoid confusion among users since the delete button will be hidden on posts and comments they are not authorized to delete.